Fix ethereum networks in schema

jtakalai opened the pull request:
also refactored the structure a bit to look more like https://linear.app/streamr/issue/ETH-184/config-package

hpihkala commented:
@mirpo How about adding a test?

harbu commented on src/connection/WsEndpoint.js:
```diff
@@ -114,8 +114,9 @@ class WsEndpoint extends EventEmitter {
-                        ws.terminate()
+                        console.error(`closing connection to ${address}...`)
+                        // force close dead of connection
+                        this._onClose(address, this.peerBook.getPeerInfo(address), 1006, 'dead-connection')
```
But we are now not terminating socket? Method `onClose` only deletes stuff from maps, objects etc. but doesn't terminate connection?

mirpo replied:
moved back, but when I was testing, it didn't help, and it doesn't trigger any event...

harbu commented on src/logic/Node.js:
```diff
@@ -339,27 +352,29 @@ class Node extends EventEmitter {
-                    streamId,
-                    node
-                })
+        return this.protocols.nodeToNode.sendSubscribe(node, streamId).then((nodeId) => {
```
If I understand correctly now subscribe messages may get resent to nodes, leading to some duplicate subscribe messages sent? But this is probably better than being unsure if subscribe has gone through, so LGTM.

mirpo replied:
Yeah, force-subscribe :)
If we'll make subscribe acknowledgement - we can bring back this check later

harbu commented on test/integration/killing-dead-connections.test.js:
```diff
@@ -0,0 +1,76 @@
+        node1.addBootstrapTracker(tracker.getAddress())
+        node2.addBootstrapTracker(tracker.getAddress())
+
+        await Promise.race([
```
Are we sure we want `race` here instead of `all`?

harbu commented on test/integration/killing-dead-connections.test.js:
```diff
@@ -0,0 +1,76 @@
+    })
+
+    it('if we find dead connection, we force close it', async () => {
+        await Promise.race([
```
Same as above.

harbu commented on test/integration/killing-dead-connections.test.js:
```diff
@@ -0,0 +1,89 @@
+    const s1 = 'stream-1'
+
+    beforeEach(async () => {
+        tracker = await startTracker(LOCALHOST, trackerPort, 'tracker')
```
Btw the core feature that is being tested here is in `WsEndpoint` right? I think you could make the test much simpler an avoid having to access members via multi-level dots (`a.b.c.d.property`) by using `startEndpoint` instead of starting tracker and nodes and subscribing and all that. For example, like in this test: https://github.com/streamr-dev/network/blob/master/test/integration/passing-address-between-ws-endpoints.test.js

harbu commented:
LGTM! 🍺

jtakalai committed:
fix: ethereum network now recognized in schema

also refactored the structure a bit to look more like https://linear.app/streamr/issue/ETH-184/config-package

jtakalai committed:
fix: change default gas pricing login for polygon

try setting a priority fee instead of hard gas price. This should make it work equally in times of high base fee.

jtakalai committed:
test: config to exclude the production gas setting

test network doesn't support EIP-1559, apparently. Maybe it should?

harbu commented on packages/client/src/StreamRegistry.ts:
```diff
@@ -95,8 +95,7 @@ export class StreamRegistry implements Context {
             StreamRegistryArtifact, this.chainProvider) as StreamRegistryContract
-        const chainId = this.config.streamRegistryChainRPC?.chainId
-        this.defaultOverrides = this.config.ethereumNetworks?.find((network) => network.chainId === chainId)?.overrides ?? {}
+        this.defaultOverrides = this.config.ethereumNetworks?.polygon?.overrides ?? {}
```
If you look at the file `ConfingBase.ts`, you will see the following entry:
```ts
streamRegistryChainRPC: {
        url: 'https://polygon-rpc.com',
        chainId: 137
}
```
Thus we leave it open to configuration whether it should use polygon network (or some other) network to deal with stream registries.

So I don't think we should hardcode `polygon` name here. I'm not sure what the intended configuration design here is, but you should probably lookup `streamRegistryChainRPC.chainId` and use that to filter the correct `etherumNetworks` entry?

jtakalai replied:
You're right it looks very much off inside that file. The idea is definitely to add an indirection there, say, streamRegistryChain would specify `polygon` as the chain to use for stream registry. I could add that indirection right now, too.

jtakalai committed:
refactor: don't use network name in code directly

jtakalai committed:
test: remove unneeded config item

this override isn't needed now that network is selected by name (dev_xdai in ConfigTest, polygon in ConfigBase)

jtakalai committed:
fix: config schema

jtakalai committed:
fix: config schema

jtakalai committed:
fix: ignore missing instead of throwing
